meteor/server/systemStatus/__tests__/systemStatus.test.ts

```diff
@@ -170,53 +160,54 @@ describe('systemStatus', () => {
 			coreVersion.patch = 999
 
 			// Change integration lib versions, simulate a patch version mismatch
-			PeripheralDevices.update(env.ingestDevice._id, {
-				$set: {
+			// Expected status is GOOD, because this should have no effect
+			await updateAndCheckSystemStatus(
+				{
 					versions: {
 						'@sofie-automation/server-core-integration': coreVersion.format(),
 					},
 				},
-			})
-			const result3: StatusResponse = await MeteorCall.systemStatus.getSystemStatus()
-
-			// Expected status is GOOD, because this should have no effect
-			const expectedStatus3 = StatusCode.GOOD
-			expect(result3).toMatchObject({
-				status: status2ExternalStatus(expectedStatus3),
-				_status: expectedStatus3,
-			})
+				env,
+				StatusCode.GOOD
+			)
 		}
 
 		// Try some silly version
-		PeripheralDevices.update(env.ingestDevice._id, {
-			$set: {
+		// Expected status is BAD, because the device expects a different version
+		await updateAndCheckSystemStatus(
+			{
 				versions: {
 					test: '0.1.2',
 				},
 			},
-		})
-		const result4: StatusResponse = await MeteorCall.systemStatus.getSystemStatus()
-
-		// Expected status is BAD, because the device expects a different version
-		const expectedStatus4 = StatusCode.BAD
-		expect(result4).toMatchObject({
-			status: status2ExternalStatus(expectedStatus4),
-			_status: expectedStatus4,
-		})
+			env,
+			StatusCode.BAD
+		)
 
 		// disableVersion check
-		PeripheralDevices.update(env.ingestDevice._id, {
-			$set: {
+		// Expected status is GOOD, as the check has been skipped
+		await updateAndCheckSystemStatus(
+			{
 				disableVersionChecks: true,
 			},
-		})
-		const result5: StatusResponse = await MeteorCall.systemStatus.getSystemStatus()
-
-		// Expected status is GOOD, as the check has been skipped
-		const expectedStatus5 = StatusCode.GOOD
-		expect(result5).toMatchObject({
-			status: status2ExternalStatus(expectedStatus5),
-			_status: expectedStatus5,
-		})
+			env,
+			StatusCode.GOOD
+		)
 	})
 })
+
+async function updateAndCheckSystemStatus(
+	setObj: object,
+	env: DefaultEnvironment,
+	expected: StatusCode
```

Comment:
`expected` on its own is a bit too vague. What is expected? Would be nice if the name gave an indication of this; for example `expectedStatusCode`

Reply:
Hàs been updated to `expectedStatusCode`.